applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_potential_flow_element.cpp

```diff
@@ -396,15 +396,18 @@ void IncompressiblePotentialFlowElement<Dim, NumNodes>::CalculateLocalSystemNorm
     // Calculate shape functions
     GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
 
-    ComputeLHSGaussPointContribution(data.vol, rLeftHandSideMatrix, data);
+    const double density_infinity = 1.0; //TODO: Read from rCurrentProcessInfo[DENSITY_INFINITY] once available
+
+    noalias(rLeftHandSideMatrix) =
```

Comment:
I think that it will be more efficient to do a unique nested loop in where you compute the LHS and the RHS at the same time. Note that this will require to firstly initialize both arrays to zero.

Reply:
Thanks for the idea! I will try to measure the building time to compare both options and make another PR for this.